stdlib/public/stubs/Availability.mm

```diff
@@ -22,72 +22,33 @@
 #import <Foundation/Foundation.h>
 #include <TargetConditionals.h>
 #include "../SwiftShims/FoundationShims.h"
-
-using namespace swift;
-
-/// Load the contents of the SystemVersion plist, looking in the appropriate
-/// place if running on the iOS simulator.
-static NSDictionary *systemVersionDictionaryFromPlist() {
-  NSString *plistPath = @"/System/Library/CoreServices/SystemVersion.plist";
-
-#if TARGET_OS_SIMULATOR
-  // When targeting the iOS simulator, look in a special location so we do
-  // not pick up the host OS version.
-  const char *simulatorRoot = getenv("IPHONE_SIMULATOR_ROOT");
-  if (!simulatorRoot) {
-    fatalError(/* flags = */ 0,
-               "Unable to check API availability: "
-               "IPHONE_SIMULATOR_ROOT not set when running under simulator");
-  }
-
-  plistPath = [NSString stringWithFormat:@"%s%@", simulatorRoot, plistPath];
-#endif
-
-  return [NSDictionary dictionaryWithContentsOfFile:plistPath];
+#import <objc/runtime.h>
+
+static NSOperatingSystemVersion getOSVersionFromFoundation() {
+  Class cls = objc_getClass("NSProcessInfo");
+  id pinfo = (id)[cls processInfo];
+  SEL sel = sel_registerName("operatingSystemVersion");
```

Comment:
Can this just be a `@selector` expression? I'd expect that to be OK if a message send is OK in the previous line.

Reply:
I'm not sure why you need to go through the runtime at all. Even if we stop importing Foundation.h, you could still make a fake protocol that declares the signatures you need for ARC, and then rely on `id` dispatch.

Reply:
Both good points. Will fix once I get things more settled and start cleaning up.